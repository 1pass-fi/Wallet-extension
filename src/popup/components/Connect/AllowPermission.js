// modules
import React from 'react'
// components
import Checkbox from 'popup/components/shared/checkBoxRedesign'

const AllowPermission = ({ handleOnClick }) => {
  return (
    <div className="mt-7.5">
      <div className="mb-6.25 font-semibold text-base leading-5 text-center tracking-finnieSpacing-wide text-indigo">
        Allow this site to:
      </div>
      <div className="flex px-9.75">
        <Checkbox
          className="check-allow"
          defaultChecked={true}
          isDisabled={true}
          greenBackround={false}
          onChange={(e) => {}}
          label="View the addresses of your permitted accounts"
          isNewDesign={true}
        />
      </div>
      <div className="flex px-9.75 mt-2">
        <Checkbox
          className="check-allow"
          defaultChecked={true}
          isDisabled={true}
          greenBackround={false}
          onChange={(e) => {}}
          label="See account balance"
          isNewDesign={true}
        />
      </div>
      <div className="flex px-9.75 mt-2">
        <Checkbox
          className="check-allow"
          defaultChecked={true}
          isDisabled={true}
          greenBackround={false}
          onChange={(e) => {}}
          label="Suggest transactions to approve"
          isNewDesign={true}
        />
      </div>
    </div>
  )
}

export default AllowPermission
