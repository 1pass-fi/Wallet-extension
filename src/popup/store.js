import { composeWithDevTools } from 'redux-devtools-extension'
import { createStore, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'

import rootReducer from './reducers/index'

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
)

export default store
