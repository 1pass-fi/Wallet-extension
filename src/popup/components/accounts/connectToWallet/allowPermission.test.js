import '@babel/polyfill'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { Provider } from 'react-redux'
import store from 'popup/store'
import React from 'react'

import AllowPermission from './allowPermission'

import { BrowserRouter as Router } from 'react-router-dom'

describe('Test for SendKoiForm component', () => {
  describe('Render without crashing', () => {
    it('renders correctly', () => {
      const { container } = render(<AllowPermission />)
      expect(container).toMatchSnapshot()
    })
  })
})
