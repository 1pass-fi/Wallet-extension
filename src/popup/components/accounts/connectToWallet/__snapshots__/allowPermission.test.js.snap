// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for SendKoiForm component Render without crashing renders correctly 1`] = `
<div>
  <div
    class="allow-permission"
  >
    <div
      class="label"
    >
      Allow this site to:
    </div>
    <div
      class="permission checked"
    >
      <input
        checked=""
        class="check-allow"
        disabled=""
        type="checkbox"
      />
      <div
        class="permission-content"
      >
        View the addresses of your permitted accounts (required)
      </div>
    </div>
    <div
      class="trust-sites"
    >
      Only connect with sites you trust.
    </div>
    <div
      class="button-line"
    >
      <button
        class="button-shared button-filled button connect"
      >
        Connect
      </button>
      <button
        class="button-shared button-outline button reject"
      >
        Reject
      </button>
    </div>
  </div>
</div>
`;
